tests/e2e/testcases/upgrade/suite.go

```diff
@@ -84,93 +86,100 @@ var _ = ginkgo.Describe("[Upgrade]", ginkgo.Ordered, func() {
+
+		output, err = commands.DescribeSubnet(subnetName)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		containsVersion2 = strings.Contains(output, binaryToVersion[utils.SoloSubnetEVMKey2])
+		gomega.Expect(containsVersion2).Should(gomega.BeFalse())
+		// the following indicates it is a custom VM
+		containsCustomVM := strings.Contains(output, "Printing genesis")
+		gomega.Expect(containsCustomVM).Should(gomega.BeTrue())
+
+		commands.DeleteSubnetConfig(subnetName)
+	})
+
+	ginkgo.It("can upgrade subnet-evm on public deployment", func() {
+		commands.CreateSubnetEvmConfig(subnetName, utils.SubnetEvmGenesisPath)
+
+		// Simulate fuji deployment
+		s := commands.SimulateFujiDeploy(subnetName, keyName, controlKeys)
+		subnetID, _, err := utils.ParsePublicDeployOutput(s)
+		gomega.Expect(err).Should(gomega.BeNil())
+		// add validators to subnet
+		nodeInfos, err := utils.GetNodesInfo()
+		gomega.Expect(err).Should(gomega.BeNil())
+		for _, nodeInfo := range nodeInfos {
+			start := time.Now().Add(time.Second * 30).UTC().Format("2006-01-02 15:04:05")
+			_ = commands.SimulateFujiAddValidator(subnetName, keyName, nodeInfo.ID, start, "24h", "20")
+		}
+		// join to copy vm binary and update config file
+		for _, nodeInfo := range nodeInfos {
+			_ = commands.SimulateFujiJoin(subnetName, nodeInfo.ConfigFile, nodeInfo.PluginDir, nodeInfo.ID)
+		}
+		// get and check whitelisted subnets from config file
+		var whitelistedSubnets string
+		for _, nodeInfo := range nodeInfos {
+			whitelistedSubnets, err = utils.GetWhilelistedSubnetsFromConfigFile(nodeInfo.ConfigFile)
+			gomega.Expect(err).Should(gomega.BeNil())
+			whitelistedSubnetsSlice := strings.Split(whitelistedSubnets, ",")
+			gomega.Expect(whitelistedSubnetsSlice).Should(gomega.ContainElement(subnetID))
+		}
+		// update nodes whitelisted subnets
+		err = utils.RestartNodesWithWhitelistedSubnets(whitelistedSubnets)
+		gomega.Expect(err).Should(gomega.BeNil())
+		// wait for subnet walidators to be up
+		err = utils.WaitSubnetValidators(subnetID, nodeInfos)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		var originalHash string
+
+		// upgrade the vm on each node
+		vmid, err := anr_utils.VMID(subnetName)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		for _, nodeInfo := range nodeInfos {
+			originalHash, err = utils.GetFileHash(filepath.Join(nodeInfo.PluginDir, vmid.String()))
+			gomega.Expect(err).Should(gomega.BeNil())
+		}
+
+		// stop network
+		commands.StopNetwork()
+
+		for _, nodeInfo := range nodeInfos {
+			_, err := commands.UpgradeVMPublic(subnetName, binaryToVersion[utils.SoloSubnetEVMKey1], nodeInfo.PluginDir)
+			gomega.Expect(err).Should(gomega.BeNil())
+		}
+
+		// This can be removed when the above is added
+		for _, nodeInfo := range nodeInfos {
+			measuredHash, err := utils.GetFileHash(filepath.Join(nodeInfo.PluginDir, vmid.String()))
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			gomega.Expect(measuredHash).ShouldNot(gomega.Equal(originalHash))
+		}
+
```

Comment:
do we want to check network is executing the new vm?

Reply:
I wanted to, but I actually had a lot of trouble getting it to work in the test with older AvalancheGo versions. I think it might be the plugin dir stuff tho. Maybe in a follow on?

Reply:
https://github.com/ava-labs/avalanche-cli/issues/569